packages/back-end/src/services/openai.ts

```diff
@@ -0,0 +1,151 @@
+import { Configuration, OpenAIApi, ChatCompletionRequestMessage } from "openai";
+import { encoding_for_model, get_encoding } from "@dqbd/tiktoken";
+import { logger } from "../util/logger";
+import { OrganizationInterface } from "../../types/organization";
+import {
+  getTokensUsedByOrganization,
+  updateTokenUsage,
+} from "../models/AITokenUsageModel";
+
+/**
+ * Snapshot of gpt-3.5-turbo from March 1st 2023. Unlike gpt-3.5-turbo, this
+ * model will not receive updates, and will be deprecated 3 months after a new
+ * version is released.
+ *
+ * We use this model to ensure behavior doesn't change while gpt-3.5-turbo is
+ * updated. Additionally, token counts will be more predictable.
+ */
+const MODEL = "gpt-3.5-turbo-0301";
+
+/**
+ * The MODEL_TOKEN_LIMIT is the maximum number of tokens that can be sent to
+ * the OpenAI API in a single request. This limit is imposed by OpenAI.
+ *
+ * Note too that very long conversations are more likely to receive incomplete
+ * replies. For example, a gpt-3.5-turbo conversation that is 4090 tokens long
+ * will have its reply cut off after just 6 tokens.
+ */
+const MODEL_TOKEN_LIMIT = 4096;
+// Require a minimum of 30 tokens for responses.
+const MESSAGE_TOKEN_LIMIT = MODEL_TOKEN_LIMIT - 30;
+
+const configuration = new Configuration({
+  apiKey: process.env.OPENAI_API_KEY || "",
+});
```

Comment:
I would move this into the `getOpenAI` function below.  If they do add validation for `apiKey` in the future, they will probably throw in the Configuration constructor.

Reply:
🤦🏾 Fixed